tfhe/src/integer/server_key/radix_parallel/tests_cases_unsigned.rs

```diff
@@ -73,6 +73,11 @@ fn overflowing_sub_under_modulus(lhs: u64, rhs: u64, modulus: u64) -> (u64, bool
     (r, q != 0)
 }
 
+fn overflowing_add_under_modulus(lhs: u64, rhs: u64, modulus: u64) -> (u64, bool) {
+    let result = lhs.wrapping_add(rhs);
+    (result % modulus, result >= modulus)
+}
```

Comment:
let's say your modulus is 2^63 + 1

I take 2^63 + 2^63 result is 2^64 or 0 in a u64, the computation overflowed but we would not see it here, am I correct ?

Maybe the sub under modulus is wrong as well 🤔

Reply:
```rust
fn overflowing_sub_under_modulus(lhs: u64, rhs: u64, modulus: u64) -> (u64, bool) {
    let overflowed = rhs > lhs;
    let result = if overflowed {
        todo!() // maybe use wrapping_sub_custom mod from the unsigned integer trait
    } else {
        lhs - rhs
    };
    (result, overflowed)
}
```

Reply:
Yeah, but these functions are written only for the tests where we don't go beyond a modulus of 2^16

Reply:
understood, could you add the assert in the two functions to avoid issues on the value ranges ? or update them so that we don't mistakenly use them in contexts where they are not meant to be used ?

Reply:
and sorry for being annoying with this but I just battled against moduli 😅 I'd rather we don't have weird surprises

Reply:
indeed should be good

Reply:
could you take a look for the sub as well ?